extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/MicrometerProcessor.java

```diff
@@ -134,6 +136,18 @@ public List<InterceptorBinding> getAdditionalBindings() {
         return UnremovableBeanBuildItem.beanTypes(METER_REGISTRY, METER_BINDER, METER_FILTER, NAMING_CONVENTION);
     }
 
+    //Not using Optional<TracerProviderBuildItem> that leads to io.quarkus.builder.ChainBuildException
+    @BuildStep
+    void ifOpentelemetryIsEnabeld(CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        if (indexBuildItem.getIndex().getClassByName(OpenTelemetry.class) == null) {
```

Comment:
It would be better to use the capabilities to test that, that's the correct pattern.

Also because you will end up with a CNFE if OpenTelemetry is not around.

Reply:
I do agree but the opentelmetry extension doesn't expose any capabilities. Should I add it ? CNFE stands for ComponentNotFoundException ? If yes, dependency is optional, so it should not happen.

Reply:
`CNFE` is `ClassNotFoundException`. The dependency is optional but you are actually referencing the class. So it's going to be a problem when the code is executed and the dependency is not around.

You could use a `DotName` and pass a string to avoid referencing the class. But really what we want here is a capability so please add one.

Reply:
Ok will do that. Maybe add a `OpentelemetryCapabilityBuildItem`. I'am only wondering where it shoulb be as the `MetricsCapabilityBuildItem` is part of quarkus-core-deployment...

Reply:
Hmmm, no, just use the standard capability feature. See `Capabilities`, `Capability` and it needs to be registered in the POM. There are plenty of examples in the codebase.

Reply:
Follow the pattern in (e.g.) the prometheus build item. We look for the class in a way that disables the build step entirely.

Reply:
Ok must have been blind, the capability does exists `Capability.OPENTELEMETRY_TRACER`.

Reply:
While I know capabilities are common, that's not the usual pattern for the micrometer extension: we skip the buildstep entirely, rather than invoking the method and then doing nothing). 

For example: 
```
@BuildSteps(onlyIf = PrometheusRegistryProcessor.PrometheusEnabled.class)
@BuildStep(onlyIf = GrpcClientSupportEnabled.class)
```

These boolean suppliers check that required classes are present AND the build-time/static config is enabled. Build steps are then entirely skipped if they don't apply.

Reply:
I think I've done what you asked.